api/src/controllers/courseController.js

```diff
@@ -36,6 +36,43 @@ const getCourses = async (req, res) => {
   })
 }
 
+// @desc    Fetch courses by category name
+// @route   GET /api/courses/:category?pageNumber=${pageNumber}
+// @access  Public
+const getCoursesByCategoryName = async (req, res) => {
```

Comment:
no need for separate function can use same getCourses with additional paramter to add where condition

Reply:
Done